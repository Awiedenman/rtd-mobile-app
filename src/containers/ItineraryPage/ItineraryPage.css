.itinerary-page {
  height: calc(100vh - 110px);
  overflow-y: scroll;
}