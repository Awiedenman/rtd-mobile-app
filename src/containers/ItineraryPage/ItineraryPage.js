import React, { Component } from 'react';
import { connect } from 'react-redux';
import ItineraryCard from '../../components/ItineraryCard/ItineraryCard';
import PropTypes from 'prop-types';
import { postFavoriteUrl, getFavoritesUrl } from '../../constants/urlGenerator';
import addFavoriteThunk from '../../thunks/addFavoriteThunk';
import getFavoritesThunk from '../../thunks/getFavoritesThunk';
import { infoCleaner, cleanStep } from '../../constants/cleanerFunctions';


import './ItineraryPage.css';

class ItineraryPage extends Component {
  constructor(props) {
    super(props)
    this.state = {
      favorite: null
    }
  }

  componentDidMount() {
    const { itinerary } = this.props;
    if (itinerary.length) {
      console.log(itinerary[0].favorite)
      this.setState({
        favorite: itinerary[0].favorite
      })
    }
  }

  addFavorite = (event) => {
    let id;
    if (this.props.itinerary.length) {
      id = this.props.itinerary[0].itinerary_id;
    }
    const { value } = event.target;
    const url = postFavoriteUrl(value, id);
    const fetchObject = {
      uid: value,
      url,
      options: {
        method: 'POST',
        headers: {
          'Content-Type': 'application/json'
        }
      }
    };
    this.props.addFavorite(fetchObject);
  }

  render() {
    const {
      itinerary,
      uid
    } = this.props;

    let startAddress;
    let endAddress;
    if (itinerary.length) {
      startAddress = itinerary[0].start_address.split(',')[0];
      endAddress = itinerary[0].end_address.split(',')[0];
    }
    const isFavorite = this.state.favorite
    ? 'favorited'
    : 'unfavorited';

    const favoriteText = isFavorite === 'favorited'
      ? ''
      : `add from: ${startAddress} to: ${endAddress} to favorites?`

    const itineraries = itinerary.map((itinerary, index) => {
      console.log(itinerary)
      return (
        <ItineraryCard
          key={index}
          itinerary={itinerary}
          uid={uid}
        />
      );
    });

    return (
      <div className='itinerary-page'>
        <div className='favorite-button-container'>
        {favoriteText}
          <button
            className={isFavorite}
            onClick={this.addFavorite}
            value={uid}
          >
          </button>
        </div>
        {itineraries}
      </div>
    );
  }
}

export const mapStateToProps = state => ({
  favorites: state.favorites,
  itinerary: state.itinerary,
  uid: state.user.uid,
});

export const mapDispatchToProps = dispatch => ({
  addFavorite: (url) => dispatch(addFavoriteThunk(url)),
  getFavorites: (url) => dispatch(getFavoritesThunk(url))
});

export default connect(mapStateToProps, mapDispatchToProps)(ItineraryPage);

ItineraryPage.propTypes = {
  favorites: PropTypes.array,
  itinerary: PropTypes.array,
  uid: PropTypes.string,
  addFavorite: PropTypes.func
};
