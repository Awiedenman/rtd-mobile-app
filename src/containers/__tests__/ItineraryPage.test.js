import React from 'react';
import { ItineraryPage } from '../ItineraryPage/ItineraryPage';
import { shallow } from 'enzyme';

describe('ItineraryPage', () => {
  it('should match snapshot', () => {
    const mockFavorites = [{
      arrival_time: "6:33am",
      departure_time: "5:51am",
      distance: "8.1 mi",
      duration: "42 mins",
      end_address: "Columbine St, Denver, CO, USA",
      favorite: false,
      itinerary_id: 59,
      start_address: "S Monaco Pkwy, Colorado, USA",
      steps: [{
        arrival_stop: "Monaco Pkwy & 29th Ave",
        arrival_time: "6:07am",
        color: "#0277bd",
        credit_name: "Regional Transportation District",
        credit_url: "http://rtd-denver.com/",
        departure_stop: "S Monaco Pkwy & Evans Ave",
        departure_time: "5:51am",
        distance: "5.5 mi",
        duration: "16 mins",
        headsign: "Central Park 65 via Monaco",
        id: 656,
        instructions: "Bus towards Central Park 65 via Monaco",
        name: "Monaco Parkway",
        num_stops: 24,
        possible_route_id: 169,
        short_name: "65",
        travel_mode: "TRANSIT",
        vehicle_type: "Bus"
      },
      {
        arrival_time: "6:33am",
        departure_time: "5:51am",
        distance: "8.1 mi",
        duration: "42 mins",
        end_address: "Columbine St, Denver, CO, USA",
        favorite: false,
        itinerary_id: 59,
        start_address: "S Monaco Pkwy, Colorado, USA",
        steps: [{
          arrival_stop: "Monaco Pkwy & 29th Ave",
          arrival_time: "6:07am",
          color: "#0277bd",
          credit_name: "Regional Transportation District",
          credit_url: "http://rtd-denver.com/",
          departure_stop: "S Monaco Pkwy & Evans Ave",
          departure_time: "5:51am",
          distance: "5.5 mi",
          duration: "16 mins",
          headsign: "Central Park 65 via Monaco",
          id: 656,
          instructions: "Bus towards Central Park 65 via Monaco",
          name: "Monaco Parkway",
          num_stops: 24,
          possible_route_id: 169,
          short_name: "65",
          travel_mode: "TRANSIT",
          vehicle_type: "Bus"
        }]
      }]
    }];

    const mockItinerary = [{
      itinerary_id: 92,
      start_address: "LoDo, Denver, CO 80202, USA",
      end_address: "650 15th St, Denver, CO 80202, USA",
      favorite: false,
      departure_time: "11:02am",
      arrival_time: "11:12am",
      duration: "10 mins",
      distance: "0.8 mi",
      title: null
    }];

    const wrapper = shallow(
      <ItineraryPage
        favorites={mockFavorites}
        user={{uid: '12'}}
        addFavorite={jest.fn()}
        itinerary={mockItinerary}
      />);

    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    test('should return an object with an itinerary', () => {
      
    });
    
  });
  
});
