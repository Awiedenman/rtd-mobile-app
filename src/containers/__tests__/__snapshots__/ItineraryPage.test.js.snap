// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItineraryPage should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItineraryPage
    addFavorite={[Function]}
    favorites={
        Array [
            Object {
              "arrival_time": "6:33am",
              "departure_time": "5:51am",
              "distance": "8.1 mi",
              "duration": "42 mins",
              "end_address": "Columbine St, Denver, CO, USA",
              "favorite": false,
              "itinerary_id": 59,
              "start_address": "S Monaco Pkwy, Colorado, USA",
              "steps": Array [
                Object {
                  "arrival_stop": "Monaco Pkwy & 29th Ave",
                  "arrival_time": "6:07am",
                  "color": "#0277bd",
                  "credit_name": "Regional Transportation District",
                  "credit_url": "http://rtd-denver.com/",
                  "departure_stop": "S Monaco Pkwy & Evans Ave",
                  "departure_time": "5:51am",
                  "distance": "5.5 mi",
                  "duration": "16 mins",
                  "headsign": "Central Park 65 via Monaco",
                  "id": 656,
                  "instructions": "Bus towards Central Park 65 via Monaco",
                  "name": "Monaco Parkway",
                  "num_stops": 24,
                  "possible_route_id": 169,
                  "short_name": "65",
                  "travel_mode": "TRANSIT",
                  "vehicle_type": "Bus",
                },
                Object {
                  "arrival_time": "6:33am",
                  "departure_time": "5:51am",
                  "distance": "8.1 mi",
                  "duration": "42 mins",
                  "end_address": "Columbine St, Denver, CO, USA",
                  "favorite": false,
                  "itinerary_id": 59,
                  "start_address": "S Monaco Pkwy, Colorado, USA",
                  "steps": Array [
                    Object {
                      "arrival_stop": "Monaco Pkwy & 29th Ave",
                      "arrival_time": "6:07am",
                      "color": "#0277bd",
                      "credit_name": "Regional Transportation District",
                      "credit_url": "http://rtd-denver.com/",
                      "departure_stop": "S Monaco Pkwy & Evans Ave",
                      "departure_time": "5:51am",
                      "distance": "5.5 mi",
                      "duration": "16 mins",
                      "headsign": "Central Park 65 via Monaco",
                      "id": 656,
                      "instructions": "Bus towards Central Park 65 via Monaco",
                      "name": "Monaco Parkway",
                      "num_stops": 24,
                      "possible_route_id": 169,
                      "short_name": "65",
                      "travel_mode": "TRANSIT",
                      "vehicle_type": "Bus",
                    },
                  ],
                },
              ],
            },
          ]
    }
    itinerary={
        Array [
            Object {
              "arrival_time": "11:12am",
              "departure_time": "11:02am",
              "distance": "0.8 mi",
              "duration": "10 mins",
              "end_address": "650 15th St, Denver, CO 80202, USA",
              "favorite": false,
              "itinerary_id": 92,
              "start_address": "LoDo, Denver, CO 80202, USA",
              "title": null,
            },
          ]
    }
    uid="12"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <ItineraryCard
          addFavorite={[Function]}
          itinerary={
                    Object {
                              "arrival_time": "11:12am",
                              "departure_time": "11:02am",
                              "distance": "0.8 mi",
                              "duration": "10 mins",
                              "end_address": "650 15th St, Denver, CO 80202, USA",
                              "favorite": false,
                              "itinerary_id": 92,
                              "start_address": "LoDo, Denver, CO 80202, USA",
                              "title": null,
                            }
          }
          uid="12"
/>,
      ],
      "className": "itinerary-page",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "class",
        "props": Object {
          "addFavorite": [Function],
          "itinerary": Object {
            "arrival_time": "11:12am",
            "departure_time": "11:02am",
            "distance": "0.8 mi",
            "duration": "10 mins",
            "end_address": "650 15th St, Denver, CO 80202, USA",
            "favorite": false,
            "itinerary_id": 92,
            "start_address": "LoDo, Denver, CO 80202, USA",
            "title": null,
          },
          "uid": "12",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <ItineraryCard
            addFavorite={[Function]}
            itinerary={
                        Object {
                                    "arrival_time": "11:12am",
                                    "departure_time": "11:02am",
                                    "distance": "0.8 mi",
                                    "duration": "10 mins",
                                    "end_address": "650 15th St, Denver, CO 80202, USA",
                                    "favorite": false,
                                    "itinerary_id": 92,
                                    "start_address": "LoDo, Denver, CO 80202, USA",
                                    "title": null,
                                  }
            }
            uid="12"
/>,
        ],
        "className": "itinerary-page",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "addFavorite": [Function],
            "itinerary": Object {
              "arrival_time": "11:12am",
              "departure_time": "11:02am",
              "distance": "0.8 mi",
              "duration": "10 mins",
              "end_address": "650 15th St, Denver, CO 80202, USA",
              "favorite": false,
              "itinerary_id": 92,
              "start_address": "LoDo, Denver, CO 80202, USA",
              "title": null,
            },
            "uid": "12",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
